router/xgress_edge/factory.go

```diff
@@ -53,7 +53,34 @@ func (factory *Factory) Enabled() bool {
 
 func (factory *Factory) BindChannel(ch channel2.Channel) error {
 	factory.ctrl = ch
-	ch.AddReceiveHandler(handler_edge_ctrl.NewHelloHandler(factory.config.Advertise, []string{"tls", "wss"}))
+
+	var parts []string
+	var hostname string
+	var supportedProtocols []string
+	var protoPorts []string
+
+	if factory.config.Advertise != "" {
+		parts = strings.Split(factory.config.Advertise, ":")
+		hostname = parts[0]
+		supportedProtocols = append(supportedProtocols, "tls")
+		protoPorts = append(protoPorts, "tls:"+parts[1])
+		pfxlog.Logger().Debugf("HelloHandler will contain hostname=[%s] supportedProtocols=%v protoPorts=%v", hostname, supportedProtocols, protoPorts)
+	}
+	if factory.config.WSAdvertise != "" {
+		parts := strings.Split(factory.config.WSAdvertise, ":")
+		if hostname != "" {
+			if hostname != parts[0] {
+				msg := fmt.Sprintf("cannot have different hostnames within multiple edge binding.advertise; got [%s] [%s]", hostname, parts[0])
+				panic(msg)
```

Comment:
I think this should be a `pfxlog.Logger.Fatalf(...)`

Reply:
done